packages/db/src/translate/granules.ts

```diff
@@ -1,11 +1,93 @@
 import Knex from 'knex';
 
-import { deconstructCollectionId } from '@cumulus/message/Collections';
+import { deconstructCollectionId, constructCollectionId } from '@cumulus/message/Collections';
+import { removeNilProperties } from '@cumulus/common/util';
 
 import { CollectionPgModel } from '../models/collection';
 import { PdrPgModel } from '../models/pdr';
-import { PostgresGranule } from '../types/granule';
+import { PostgresGranule, PostgresGranuleRecord } from '../types/granule';
 import { ProviderPgModel } from '../models/provider';
+import { FilePgModel } from '../models/file';
+import { translatePostgresFileToApiFile } from './file';
+import { getExecutionArnsByGranuleCumulusId } from '../lib/execution';
+
+/**
+ * Generate an API Granule object from a Postgres Granule with associated Files.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} granulePgRecord
+ *   Granule from Postgres
+ * @param {Knex | Knex.Transaction} knexOrTransaction
+ *   Knex client for reading from RDS database
+ * @param {Object} collectionPgModel - Instance of the collection database model
+ * @param {Object} pdrPgModel - Instance of the pdr database model
+ * @param {Object} providerPgModel - Instance of the provider database model
+ * @param {Object} filePgModel - Instance of the file database model
+ * @param {Object} executionPgModel - Instance of the execution database model
+ * @returns {Object} An API Granule with associated Files
+ */
+export const translatePostgresGranuleToApiGranule = async (
+  granulePgRecord: PostgresGranuleRecord,
+  knexOrTransaction: Knex | Knex.Transaction,
+  collectionPgModel = new CollectionPgModel(),
+  pdrPgModel = new PdrPgModel(),
+  providerPgModel = new ProviderPgModel(),
+  filePgModel = new FilePgModel()
+): Promise<AWS.DynamoDB.DocumentClient.AttributeMap> => {
+  const collection = await collectionPgModel.get(
+    knexOrTransaction, { cumulus_id: granulePgRecord.collection_cumulus_id }
+  );
+  const files = await filePgModel.search(
+    knexOrTransaction, { granule_cumulus_id: granulePgRecord.cumulus_id }
+  );
+  const executionArns = await getExecutionArnsByGranuleCumulusId(
+    knexOrTransaction,
+    granulePgRecord.cumulus_id
+  );
+
+  let pdr;
+
+  if (granulePgRecord.pdr_cumulus_id) {
+    pdr = await pdrPgModel.get(
+      knexOrTransaction, { cumulus_id: granulePgRecord.pdr_cumulus_id }
+    );
+  }
+
+  let provider;
+
+  if (granulePgRecord.provider_cumulus_id) {
+    provider = await providerPgModel.get(
+      knexOrTransaction, { cumulus_id: granulePgRecord.provider_cumulus_id }
+    );
+  }
+
+  return removeNilProperties(({
+    granuleId: granulePgRecord.granule_id,
```

Comment:
Nit: consider alpha sorted keys, here and in the test.

Reply:
https://github.com/nasa/cumulus/pull/2415/commits/a623948037ff64737a3d173ef661003b83d9cd19